arviz/data/inference_data.py

```diff
@@ -76,36 +89,59 @@ def __init__(self, **kwargs):
 
         """
         self._groups = []
-        key_list = [key for key in SUPPORTED_GROUPS if key in kwargs]
+        self._groups_warmup = []
+        save_warmup = kwargs.pop("save_warmup", False)
+        key_list = [key for key in SUPPORTED_GROUPS_ALL if key in kwargs]
         for key in kwargs:
-            if key not in SUPPORTED_GROUPS:
+            if key not in SUPPORTED_GROUPS_ALL:
                 key_list.append(key)
                 warnings.warn(
                     "{} group is not defined in the InferenceData scheme".format(key), UserWarning
                 )
         for key in key_list:
             dataset = kwargs[key]
+            dataset_warmup = None
             if dataset is None:
                 continue
+            elif isinstance(dataset, (list, tuple)):
+                dataset, dataset_warmup = kwargs[key]
             elif not isinstance(dataset, xr.Dataset):
                 raise ValueError(
                     "Arguments to InferenceData must be xarray Datasets "
                     "(argument '{}' was type '{}')".format(key, type(dataset))
                 )
-            setattr(self, key, dataset)
-            self._groups.append(key)
+            if not key.startswith("_warmup_"):
```

Comment:
More pickiness. We should move `"_warmup_"` to be a global as well so we can change easily and its obvious its the same thing everywhere.

I can help with all these little nitpicks if you like. Just let me know and I can push to PR.

Reply:
Feel free to edit. (Others can too)